web/src/features/exchanges/ExchangeArrow.tsx

```diff
@@ -26,124 +23,82 @@ function ExchangeArrow({
   map,
   colorBlindMode,
 }: ExchangeArrowProps) {
-  const mapZoom = map.getZoom();
-  const colorBlindModeEnabled = colorBlindMode;
-  const absFlow = Math.abs(data.netFlow ?? 0);
   const { co2intensity, lonlat, netFlow, rotation, key } = data;
-  const setIsMoving = useSetAtom(mapMovingAtom);
   if (!lonlat) {
     return null;
   }
 
-  useEffect(() => {
-    const cancelWheel = (event: Event) => event.preventDefault();
-    const exchangeLayer = document.querySelector('#exchange-layer');
-    if (!exchangeLayer) {
-      return;
-    }
-    exchangeLayer.addEventListener('wheel', cancelWheel, {
-      passive: false,
-    });
-    return () => exchangeLayer.removeEventListener('wheel', cancelWheel);
-  }, []);
-
-  const imageSource = useMemo(() => {
-    const prefix = colorBlindModeEnabled ? 'colorblind-' : '';
-    const intensity = quantizedCo2IntensityScale(co2intensity);
-    const speed = quantizedExchangeSpeedScale(Math.abs(netFlow));
-    return resolvePath(`images/arrows/${prefix}arrow-${intensity}-animated-${speed}`)
-      .pathname;
-  }, [colorBlindModeEnabled, co2intensity, netFlow]);
-
+  const absFlow = Math.abs(netFlow ?? 0);
+  // Don't render if the flow is very low ...
+  if (absFlow < 1) {
+    return null;
+  }
+  const mapZoom = map.getZoom();
   const projection = map.project(lonlat);
-
   const transform = {
     x: projection.x,
     y: projection.y,
     k: 0.04 + (mapZoom - 1.5) * 0.1,
     r: rotation + (netFlow > 0 ? 180 : 0),
   };
 
-  // Don't render if the flow is very low ...
-  if (absFlow < 1) {
-    return null;
-  }
-
-  // ... or if the arrow would be very tiny ...
-  if (transform.k < 0.1) {
-    return null;
-  }
-
-  // ... or if it would be rendered outside of viewport.
-  if (transform.x + 100 * transform.k < 0) {
+  if (
+    // or if the arrow would be very tiny
+    transform.k < 0.1 ||
+    // or if it would be rendered outside of viewport.
+    transform.x + 100 * transform.k < 0 ||
+    transform.x - 100 * transform.k > viewportWidth ||
+    transform.y - 100 * transform.k > viewportHeight
+  ) {
     return null;
   }
 
-  if (transform.y + 100 * transform.k < 0) {
-    return null;
-  }
+  const setIsMoving = useSetAtom(mapMovingAtom);
 
-  if (transform.x - 100 * transform.k > viewportWidth) {
-    return null;
-  }
+  useEffect(() => {
```

Comment:
Doesn't this break the [Rules of Hooks](https://legacy.reactjs.org/docs/hooks-rules.html), as the hook is only conditionally rendered? Guess this was also a problem before though, so we should probably use the linting plugin for hooks: https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks

Reply:
I do not know but since the renders should not change between the hours themselves and we are re rendering all hours I think we should be safe, but I'll look into it.

Reply:
I'll leave it as is now and then I'll open a new PR with the new eslint plugin and fix everything that needs to be fixed in there.